packages/ketcher-core/src/application/editor/actions/rgroupAttachmentPoint.ts

```diff
@@ -0,0 +1,126 @@
+import assert from 'assert';
+import { ReStruct } from 'application/render';
+import { Struct } from 'domain/entities';
+import {
+  RGroupAttachmentPointAdd,
+  RGroupAttachmentPointRemove,
+} from '../operations';
+import { Action } from './action';
+import { fromAtomsAttrs } from '.';
+
+export function fromRGroupAttachmentPointUpdate(
+  restruct: ReStruct,
+  atomId: number,
+  attpnt,
+) {
+  const action = new Action();
+  action.mergeWith(fromRGroupAttachmentPointsDeletionByAtom(restruct, atomId));
+  action.mergeWith(fromRGroupAttachmentPointAddition(restruct, attpnt, atomId));
+  return action;
+}
+
+function fromRGroupAttachmentPointAddition(
+  restruct: ReStruct,
+  attpnt,
+  atomId: number,
+) {
+  const action = new Action();
+  switch (attpnt) {
+    case 1:
+      action.addOp(
+        new RGroupAttachmentPointAdd({
+          atomId,
+          attachmentPointType: 'primary',
+        }),
+      );
+      break;
+
+    case 2:
+      action.addOp(
+        new RGroupAttachmentPointAdd({
+          atomId,
+          attachmentPointType: 'secondary',
+        }),
+      );
+      break;
+
+    case 3:
+      action.addOp(
+        new RGroupAttachmentPointAdd({
+          atomId,
+          attachmentPointType: 'primary',
+        }),
+      );
+      action.addOp(
+        new RGroupAttachmentPointAdd({
+          atomId,
+          attachmentPointType: 'secondary',
+        }),
+      );
+      break;
+
+    default:
+      break;
+  }
+  return action.perform(restruct);
+}
+
+function fromRGroupAttachmentPointsDeletionByAtom(
+  restruct: ReStruct,
+  atomId: number,
+) {
+  const action = new Action();
+  const attachmentPointsToDelete =
+    restruct.molecule.getRGroupAttachmentPointsByAtomId(atomId);
+  attachmentPointsToDelete.forEach((rgroupAttachmentPointId) => {
+    action.addOp(new RGroupAttachmentPointRemove(rgroupAttachmentPointId));
+  });
+  return action.perform(restruct);
+}
+
+export function fromRGroupAttachmentPointDeletion(
+  restruct: ReStruct,
+  id: number,
+) {
+  const { atomId, newAttpnt } = getNewAtomAttpnt(restruct.molecule, id);
```

Comment:
I didn't catch why we need to get new attachment point during the deletion. Could you please explain?

Reply:
I've commented meanings of values of `attpnt` here https://github.com/epam/ketcher/pull/2961, will add type annotation to `attpnt` in https://github.com/epam/ketcher/issues/2962.

As you can see in https://github.com/epam/ketcher/pull/2961 `attpnt` has five values 
```typescript
/**
   *  Values can be `0 | 1 | 2 | 3 | null`.
   * `1` - has a `primary` R-Group Attachment Point
   * `2` - has a `secondary` R-Group Attachment Point
   * `3` - has a `primary` and a `secondary` R-Group Attachment Points
   * `null` and `0` both mean the atom has no R-Group Attachment Points
   */
```
Imagine you have an atom whose `attpnt` is `3`, which means it has both a `primary` and a `secondary` R-Group Attachment Points. Now you want to delete one of the attachment points. 
- If you delete `primary`, then `getNewAtomAttpnt` will give you a new attpnt `2`. 
- If you delete `secondary`, then `getNewAtomAttpnt` will give you a new attpnt `1`.